docs/general/dql/selects.rst

```diff
@@ -841,6 +841,50 @@ Object property can also be addressed in the :ref:`where clause
     +------------+----------------------------------------------------------------------+
     SELECT 1 row in set (... sec)
 
+For selecting records having values with empty or unassigned objects, like::
+
+    cr> insert into locations (id, name, position, kind, inhabitants) values
+    ... (99, 'Voidspace', 0, 'Milliways', {});
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty objects, is::
+
+    cr> select count(*) from locations where inhabitants is null;
+    +----------+
+    | count(*) |
+    +----------+
+    |       12 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+.. Hidden: drop record again
+
+    cr> delete from locations where id=99;
+    DELETE OK, 1 row affected  (... sec)
+
+For determining the number of items in an object, you can use the
+:ref:`object_keys <scalar-object_keys>` and :ref:`array_length
+<scalar-array_length>` scalar functions::
+
+    cr> select count(*) from locations where
+    ... array_length(object_keys(inhabitants), 1) = 2;
+    +----------+
+    | count(*) |
+    +----------+
+    |        1 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+.. NOTE::
+
+    ``array_length(object_keys(inhabitants), 1)`` will return ``NULL`` on empty
+    objects.
+
```

Comment:
```suggestion
```

Seems unrelated and doesn't explain why this is the case.

Reply:
Maybe not good explained, but it worth mentioning somewhere (maybe better at the `array_length` documentation), that `array_length(some_array, 1)` will return `NULL` on empty arrays while someone may expect `0`.

Reply:
@mfussenegger: Yes, I will rework this patch and remove unrelated and confusing examples. @seut: I agree and will amend the other patch #12688 correspondingly.